jobs/upload-brew.Jenkinsfile

```diff
@@ -0,0 +1,128 @@
+// kola parallelism from that. We leave 512M for overhead and VMs are at most
+// 1.5G each (in general; root reprovisioning tests require 4G which is partly
+// why we run them separately).
+// XXX: https://github.com/coreos/coreos-assembler/issues/3118 will make this
+// cleaner
+def ncpus = ((cosa_memory_request_mb - 512) / 1536) as Integer
+
+echo "Waiting for build-upload-to-brew lock"
+currentBuild.description = "${build_description} Waiting"
+def stream_info = pipecfg.streams[params.STREAM]
+
+lock(resource: "build-upload-to-brew") {
+    timeout(time: 240, unit: 'MINUTES') {
+    cosaPod(cpu: "${ncpus}",
+            memory: "${cosa_memory_request_mb}Mi",
+            image: cosa_img,
+            serviceAccount: "jenkins") {
+    try {
+        stage('Fetch Metadata') {
+            def ref = pipeutils.get_source_config_ref_for_stream(pipecfg, params.STREAM)
+            def variant = stream_info.variant ? "--variant ${stream_info.variant}" : ""
+            def s3_stream_dir = pipeutils.get_s3_streams_dir(pipecfg, params.STREAM)
+            pipeutils.shwrapWithAWSBuildUploadCredentials("""
+                cosa init --branch ${ref} ${variant} ${pipecfg.source_config.url}
+                cosa buildfetch --build=${params.VERSION} \
+                    --arch=${params.ARCH} --url=s3://${s3_stream_dir}/builds \
+                    --aws-config-file \${AWS_BUILD_UPLOAD_CONFIG} \
+                    --artifact=qemu \
+                    --file "coreos-assembler-config-git.json" 
+            """)
+        }
+        stage('Brew Upload') {
+            withCredentials([
+                file(credentialsId: 'brew-cfg', variable: 'BREW'),
+                     file(credentialsId: 'brew-krb-config', variable: 'KRB5'),
+                     file(credentialsId: 'brew-ca', variable: 'CA'),
+                     file(credentialsId: 'brew-keytab', variable: 'KEYTAB'),
+             ]) {
+                shwrap("""
+                    cp $BREW /etc/koji.conf
+                    cp $KRB5 /etc/krb5.conf
+                    cp $KEYTAB /tmp/keytab
+                    cp $CA /etc/pki/brew/brew-ca
+                    coreos-assembler koji-upload \
+                        upload --reserve-id \
+                        --keytab /tmp/keytab \
+                        --build ${version} \
+                        --retry-attempts 6 \
+                        --buildroot builds \
+                        --owner ${brew_principle} \
+                        --profile ${brew_profile} \
+                        --tag rhaos-${params.STREAM}-rhel-build
+                """)
+            }
+        }
+        currentBuild.result = 'SUCCESS'
+
+    // main timeout and try {} finish here
+    } catch (e) {
+             currentBuild.result = 'FAILURE'
+             throw e
+         } finally {
+             def color
+             def message = "[${params.VERSION}][${params.ARCH}]"
```

Comment:
probably want the stream name in here too and also the name of the job. we want this message to be similar to the kola jobs.. i.e. something like:

```
:fcos: :gcp: kola-gcp [#394](https://jenkins-fedora-coreos-pipeline.apps.ocp.fedoraproject.org/job/kola-gcp/394/) [rawhide][x86_64] (39.20230313.91.0)
```